packages/task/src/browser/task-contribution.ts

```diff
@@ -70,15 +70,22 @@ export class TaskProviderRegistry {
     }
 
     /** Registers the given Task Provider to return Task Configurations of the specified type. */
-    register(type: string, provider: TaskProvider): Disposable {
-        this.providers.set(type, provider);
+    register(type: string, provider: TaskProvider, handle?: number): Disposable {
+        const key = handle === undefined ? type : `${type}::${handle}`;
+        this.providers.set(key, provider);
         return {
-            dispose: () => this.providers.delete(type)
+            dispose: () => this.providers.delete(key)
         };
     }
 
-    getProvider(type: string): TaskProvider | undefined {
-        return this.providers.get(type);
+    getProvider(type: string): TaskProvider[] {
```

Comment:
Does the return type not break API?
If anyone called `getProvider` they expect either a `TaskProvider` or `undefined` but now we return an array.
I assume that is why we have the method `getProviders` which already returns an array.

Reply:
@vince-fugnitto Thanks for the review, code adjusted and those providers return to their original values.